qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/AllergyPage.java

```diff
@@ -70,4 +71,9 @@ public void removeNoKnownAllergy() {
 	public String getPageUrl() {
 		return "/allergyui/allergies.page";
 	}
+	
+	public ClinicianFacingPatientDashboardPage clickOnReturn() {
+		clickOn(RETURN_BUTTON);
+		return new ClinicianFacingPatientDashboardPage(this);
+	}
```

Comment:
do we really need this method in AllergyPage?

Reply:
@kdaud,  The test starts from the patient dashbaord and basically it should return to the same when it is done. So for the test to return to the patient dashboard, it should use the return button which resides in the AllergyPage. So that method is added such that the return button  could be of use when it comes to getting back to the patient dashboard.

Reply:
isn't  [clickOnReturn()](https://github.com/openmrs/openmrs-contrib-qaframework/blob/c369439ed5e0b083b47f9a7dcf6df519a68565a5/qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/AddOrEditAllergyPage.java#L86-L89) sufficient enough to perform the task?

Reply:
> isn't [clickOnReturn()](https://github.com/openmrs/openmrs-contrib-qaframework/blob/c369439ed5e0b083b47f9a7dcf6df519a68565a5/qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/AddOrEditAllergyPage.java#L86-L89) sufficient enough to perform the task?

If we only use clickOnReturn as a void method, it will return a null pointer exception in case the test wants to return to the patient dashboard.

Reply:
what is the return type of [clickOnReturn()](https://github.com/openmrs/openmrs-contrib-qaframework/blob/c369439ed5e0b083b47f9a7dcf6df519a68565a5/qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/AddOrEditAllergyPage.java#L86-L89)?

Reply:
> what is the return type of [clickOnReturn()](https://github.com/openmrs/openmrs-contrib-qaframework/blob/c369439ed5e0b083b47f9a7dcf6df519a68565a5/qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/AddOrEditAllergyPage.java#L86-L89)?

@kdaud I don't know if I got you right on this. However, clickOnReturn() was not native method in the AllergyPage, I just introduced it and I wanted it to return the dashboard page. I don't know if I have appropriately answered your question.

Reply:
why not use the existing [method](https://github.com/openmrs/openmrs-contrib-qaframework/blob/c369439ed5e0b083b47f9a7dcf6df519a68565a5/qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/AddOrEditAllergyPage.java#L86-L89) instead?

Reply:
> why not use the existing [method](https://github.com/openmrs/openmrs-contrib-qaframework/blob/c369439ed5e0b083b47f9a7dcf6df519a68565a5/qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/AddOrEditAllergyPage.java#L86-L89) instead?

@kdaud have you looked at the flow of the test properly? Basically the test can not use that existing method in the addOrEditAllergyPage because at the point of deleting an allergy after which it returns to the patient dashboard, it is not interacting with the addOrEditAllergyPage at all, It is actually at the AllergyPage which is the allergies board page. So for the test to be able to return to the dashboard page it has to hit the return button that resides on the AllergyPage. That's why I had to create that method in the AllergyPage.

Reply:
@Irenyak1 the method being suggested in AllergyPage is a duplicate of what is already existing in the child class(AddOrEditAllergyPage) where the steps are being executed.

Have you tried to use the method (in the test implementation ) am referring to and its not working out?

```js
@And("the system deletes an allergy from the allergies table")
 public void systemDeletesAllergy() {
     assertTrue(allergyPage.getAllergyStatus().contains("Unknown"));
    dashboardPage = addOrEditAllergyPage.clickReturn();
}
```

Reply:
> @Irenyak1 the method being suggested in AllergyPage is a duplicate of what is already existing in the child class(AddOrEditAllergyPage) where the steps are being executed.
> 
> Have you tried to use the method (in the test implementation ) am referring to and its not working out?
> 
> ```js
> @And("the system deletes an allergy from the allergies table")
>  public void systemDeletesAllergy() {
>      assertTrue(allergyPage.getAllergyStatus().contains("Unknown"));
>      dashboardPage = addOrEditAllergyPage.clickReturn();
> }
> ```

Oh yeah @kdaud  that can work. Let me just do that. Thanks.